test/server/options_impl_test.cc

```diff
@@ -19,37 +19,45 @@ using testing::HasSubstr;
 
 namespace Envoy {
 
-namespace {
-
-// Do the ugly work of turning a std::string into a char** and create an OptionsImpl. Args are
-// separated by a single space: no fancy quoting or escaping.
-std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
-  std::vector<std::string> words = TestUtility::split(args, ' ');
-  std::vector<const char*> argv;
-  for (const std::string& s : words) {
-    argv.push_back(s.c_str());
+class OptionsImplTest : public testing::Test {
+
+public:
+  // Do the ugly work of turning a std::string into a char** and create an OptionsImpl. Args are
+  // separated by a single space: no fancy quoting or escaping.
+  std::unique_ptr<OptionsImpl> createOptionsImpl(const std::string& args) {
+    std::vector<std::string> words = TestUtility::split(args, ' ');
+    std::vector<const char*> argv;
+    for (const std::string& s : words) {
+      argv.push_back(s.c_str());
+    }
+    return std::make_unique<OptionsImpl>(argv.size(), argv.data(),
+                                         [](uint64_t, uint64_t, bool) { return "1"; },
+                                         spdlog::level::warn);
   }
-  return std::make_unique<OptionsImpl>(
-      argv.size(), argv.data(), [](uint64_t, uint64_t, bool) { return "1"; }, spdlog::level::warn);
-}
 
-} // namespace
+  const std::vector<std::pair<std::string, spdlog::level::level_enum>>&
+  parseComponentLogLevels(const std::unique_ptr<OptionsImpl>& options,
```

Comment:
Is it worth doing this indirection here vs. just having parseComponentLogLevels be a static function that returns a map? I think that would be simpler and clearer? Then I think you can revert all the TEST_F changes potentially and just have mostly tests for the static helper and a single real option test with some real values to check? Up to you.

Reply:
I think will leave it like this.